projects/schematics/src/migrations/3_0/constructor-deprecations/data/checkout.service.migration.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  AUTH_SERVICE,
+  SPARTACUS_CORE,
+  CHECKOUT_SERVICE,
+  USER_ID_SERVICE,
+  STORE,
+  NGRX_STORE,
+  ACTIVE_CART_SERVICE,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const CHECKOUT_SERVICE_MIGRATION: ConstructorDeprecation = {
+  // projects/core/src/checkout/facade/checkout.service.ts
+  class: CHECKOUT_SERVICE,
+  importPath: SPARTACUS_CORE,
+  deprecatedParams: [
+    {
+      className: STORE,
+      importPath: NGRX_STORE,
+    },
+    {
+      className: AUTH_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: ACTIVE_CART_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+  ],
+  removeParams: [
+    {
+      className: AUTH_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+  ],
+  addParams: [
```

Comment:
To make this migration work we need to move `UserIdService` in the service to be the last constructor parameter.